src/components/Header/index.tsx

```diff
@@ -36,18 +44,49 @@ export default function Header({
   title,
   goBackArrow,
   roomCode,
+  participants,
   timer,
   settingsPage,
   infoPage,
 }: HeaderProps) {
   const { room } = useGlobalContext();
   const navigateTo = useNavigate();
   const [warningVisibility, setWarningVisibility] = useState<boolean>(false);
+  const [playerListVisibility, setPlayerListVisibility] =
+    useState<boolean>(false);
 
   const seconds = timer / 1000;
   const timerColor = seconds < 3 ? 'red' : 'white';
   const formattedTimer = `${seconds.toFixed(1)}s`;
 
+  const playerList = useMemo(
+    () => (
+      <>
+        {room.playerList.map((p, i) => (
+          <ListItem key={i}>
+            <PlayerInfo>
+              <AvatarDiv>
+                <Avatar seed={p.avatarSeed} />
+              </AvatarDiv>
+              <Name>{p.nickname}</Name>
+            </PlayerInfo>
+            <Kick onClick={() => kickPlayer(p)}>Expulsar</Kick>
+          </ListItem>
+        ))}
+      </>
+    ),
+    [room.playerList]
+  );
+
+  const kickPlayer = (p: Player) => {
+    const socket = SocketConnection.getInstance();
+    socket &&
```

Comment:
Acho que vale a pena adicionar um condicional pra this.socket dentro da função push, da mesma maneira que a gente faz pra função disconnect. Só pra se proteger no caso do header renderizar sem a conexão ativa.